pkg/agent/agent.go

```diff
@@ -85,6 +92,8 @@ type NimbessAgent struct {
 	Pipelines map[string]*NimbessPipeline
 	// MetaPipelines contains a map of network name to abstract pipelines
 	MetaPipelines map[string]*NimbessPipeline
+    l2fib map[string]L2FIBEntry
```

Comment:
something wrong with the spacing here, can you please run:
gofmt -w -s pkg/* && gofmt -w -s cmd/*

Reply:
In fact what we could do is make the Meta pipeline switch module hold the L2FIB (it already has a map for that), then all of the port pipelines created from this Meta pipeline could set their L2FIB as a pointer to the Meta Switch L2FIB. Then you only need to modify the FIB of the meta switch.

Reply:
Looks like you are using hard tabs while I have mine set-up for soft-tab which the editor fills with spaces after that. I use hard-tap only for the linux source where it is a code style requirement (no spaces for indentation).

Reply:
> In fact what we could do is make the Meta pipeline switch module hold the L2FIB (it already has a map for that), then all of the port pipelines created from this Meta pipeline could set their L2FIB as a pointer to the Meta Switch L2FIB. Then you only need to modify the FIB of the meta switch.

I need to think on this as this will require reorganizing quite a few things - f.e. we may need to spin up an l2fib notification thread per metapipeline instead of a centralized agent thread and manage notification channels accordingly. 

For now, I will limit the mac fib updates so that they are limited to the same metakey - this takes care of network isolation.